Работа с балансами

nue-01 opened the pull request:
Делал от ветки BJS2-41970 (были нужные изменения в ней)
Пока не влили BJS2-41970, ПР направил в эту ветку (чтобы корректно изменения отображались)

nue-01 committed:
BJS2-41974-Работа с балансами.
Написал эндпоинты для работы с балансами, интеграционные и юнит тесты.

nue-01 committed:
Merge branch 'feature/BJS2-41970' into feature/BJS2-41974

# Conflicts:
#	src/main/resources/db/changelog/db.changelog-master.yaml
#	src/test/java/faang/school/accountservice/controller/account/AccountControllerTest.java
#	src/test/java/faang/school/accountservice/service/account/AccountServiceTest.java
#	src/test/java/faang/school/accountservice/validator/account/AccountValidatorTest.java

nue-01 committed:
BJS2-41974-Работа с балансами.
Решил мердж конфликты

Shakhaev commented on src/main/java/faang/school/accountservice/repository/account/BalanceRepository.java:
```diff
@@ -0,0 +1,10 @@
+
+import java.util.Optional;
+
+public interface BalanceRepository extends JpaRepository<Balance, Long> {
```
Привет! Здесь не хватает аннотации @Repostitory

Shakhaev commented on src/main/java/faang/school/accountservice/service/account/BalanceService.java:
```diff
@@ -0,0 +1,43 @@
+        Account account = accountRepository.findById(accountId)
+                .orElseThrow(() -> new EntityNotFoundException("Account doesn't exist"));
+        Balance balance = new Balance().setAccount(account);
+        return balanceMapper.toDto(balanceRepository.save(balance));
```
Думаю, здесь лучше логику сохранения вынести чуть выше, а уже ниже оставить возвращение маппинга в DTO, поскольку в будущем наверняка нужно будет публиковать событие для уведомления юзера о создании баланса для счета. Если мы внесем логику публикации выше, чем сохранение баланса в репозитории, то при откате транзакции у нас будет казус: уведомление отправлено, а счет не создан.

nue-01 committed:
BJS2-41974-Работа с балансами.
Поставил аннотацию repository; в сервисе отделил операции сохранения и возвращения записи.

ZhubanyshZh reviewed (commented):
@DaniyarDostiev Привет посмотрел нет замечаний

MiraGlaceon commented on src/main/java/faang/school/accountservice/controller/account/BalanceController.java:
```diff
@@ -0,0 +1,35 @@
+public class BalanceController {
+    private final BalanceService balanceService;
+
+    @GetMapping("/{accountId}")
```
Лучше /accounts/{accountId}/balances сделать. Ты находишь сначала ресурс, потом смотришь его баланс по логике в url

nue-01 committed:
BJS2-41974-Работа с балансами.
Изменил url'ы в контроллере работы с балансами.

nue-01 committed:
BJS2-41974-Работа с балансами.
Изменил url'ы в инт. тестах работы с балансами.
